### PR TITLE
test: unit test for vmware_tools connection

### PR DESCRIPTION
**Pull request by Akasurde**
##### SUMMARY
Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
test/units/plugins/connection/test_vmware_tools.py

### CODE REVIEW

**Comment by Akasurde** [comment]
cc @jamescassell Could you please review this ? Thanks.

File: test/units/plugins/connection/test_vmware_tools.py [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+                'ansible_vmware_guest_path': 'ha-datacenter/vm/VM_0001',
+                'ansible_vmware_tools_user': 'root',
+            },
+            (
+                "vmware_tools setting: vm_password"
+            ),
+        ),
+    )
+
+    @pytest.mark.parametrize('options, expected', ((o, e) for o, e in OPTIONS_DATA))
```
Comment (jamescassell) [inline test/units/plugins/connection/test_vmware_tools.py:77]:
```suggestion
    @pytest.mark.parametrize('options, expected', ((o, e) for o, e in self.OPTIONS_DATA))
```

shot in the dark

**Review by jamescassell — APPROVED** [review]
Overall, it looks good to me.  Testing is not my strength, though I can mostly follow what it's testing.

@dericcrago does this test look good to you?

**Comment by dericcrago** [comment]
@jamescassell - you're doing better than me. ;)

@Akasurde - could you provide a brief summary of your strategy and how this works? I'm not real clear on what is *actually* being tested with the use of the mocks.

**Comment by Akasurde** [comment]
> @Akasurde - could you provide a brief summary of your strategy and how this works? I'm not real clear on what is _actually_ being tested with the use of the mocks.

@dericcrago Thanks for the vmware_tools connection plugin. This is just a unit test for checking basic functionality of vmware_tools connection. We are working (2.9) on CI environment based upon real vCenter environment which will test this connection plugin as a part of integration tests. let me know if you need any other information.

Thanks.

**Pull request closed without merging**